google-cloud-firestore/src/main/java/com/google/cloud/firestore/FirestoreImpl.java

```diff
@@ -212,21 +220,37 @@ void getAll(
       @Nullable FieldMask fieldMask,
       @Nullable ByteString transactionId,
       final ApiStreamObserver<DocumentSnapshot> apiStreamObserver) {
-
     ResponseObserver<BatchGetDocumentsResponse> responseObserver =
         new ResponseObserver<BatchGetDocumentsResponse>() {
           int numResponses = 0;
           boolean hasCompleted = false;
 
           @Override
-          public void onStart(StreamController streamController) {}
+          public void onStart(StreamController streamController) {
+            getTraceUtil()
+                .currentSpan()
+                .addEvent(
+                    TraceUtil.SPAN_NAME_BATCH_GET_DOCUMENTS + ": Start",
+                    Collections.singletonMap("numDocuments", documentReferences.length));
+          }
 
           @Override
           public void onResponse(BatchGetDocumentsResponse response) {
             DocumentReference documentReference;
             DocumentSnapshot documentSnapshot;
 
             numResponses++;
+            if (numResponses == 1) {
+              getTraceUtil()
+                  .currentSpan()
+                  .addEvent(TraceUtil.SPAN_NAME_BATCH_GET_DOCUMENTS + ": First response received");
+            } else if (numResponses % 100 == 0) {
+              // To reduce the size of traces, we only register one event for every 100 responses
+              // that we receive.
+              getTraceUtil()
+                  .currentSpan()
+                  .addEvent(TraceUtil.SPAN_NAME_BATCH_GET_DOCUMENTS + ": Received 100 responses");
```

Comment:
Nit: Do we want to log the number of the documents responses to the event so the events won't be the same?
 .addEvent(TraceUtil.SPAN_NAME_BATCH_GET_DOCUMENTS + ": Received " +  numResponses +" responses");

Reply:
Yeah that's a good idea. Done.